### PR TITLE
[docs]: Add Podman  Guide

### LINKED ISSUE #2203: Podman - Podman Documentation
# Podman Documentation
child of #760 

This issue will involve creating a guide for Podman usage.

## Included
- [x] Add a guide page for Podman with instructions on how to set it up, how to configure earthly to use it, known limitations, and troubleshooting.
 - [ ] Update [Installation Instructions](https://earthly.dev/get-earthly) and [Basics](https://docs.earthly.dev/basics) to reference the guide.

### PR DESCRIPTION
**Pull request by r-LaForge**
Relates to #2203 

you can view the changes here: https://docs.earthly.dev/~/revisions/rxxd8cpoimIreO7T6Pbt/docs/guides/podman

### CODE REVIEW

File: docs/guides/podman.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+When earthly starts a check is done to determine what frontend is available.
+By default, earthly will attempt to use docker and then fall back to podman.
+If you wish to change the behavior of the startup check, run the following command:
+
+```bash
+# Configure earthly to use podman
+> earthly config global.container_frontend podman-shell
+
+# Configure earthly to use docker
+> earthly config global.container_frontend docker-shell
```
Comment (vladaionescu) [inline docs/guides/podman.md:24]:
```suggestion
earthly config global.container_frontend podman-shell

# Configure earthly to use docker
earthly config global.container_frontend docker-shell
```

We've used this convention on other pages: no terminal prompt, to make it easier to copy-paste.

File: docs/guides/podman.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+...
+graphDriverName: overlay  # or something similar
+...
+```
+
+### Mac: docker-credential-desktop: executable file not found in $PATH
+This error typically occurs when switching from docker desktop to podman without docker installed.
+There may be a lingering configuration file that will be read by the attachable used to authenticate calls to buildkit.
+
+To fix this issue, try removing or renaming the `!/.docker/config.json` file.
```
Comment (vladaionescu) [inline docs/guides/podman.md:104]:
```suggestion
To fix this issue, try removing or renaming the `~/.docker/config.json` file.
```

**Review by vladaionescu — COMMENTED** [review]
This looks good in general! I left some relatively minor comments below...

We should also include this in our installation instructions. I wonder if this page should simply be linked from https://earthly.dev/get-earthly and from https://docs.earthly.dev/docs/misc/alt-installation.

The one important thing I'm wondering is what would be a good structure for the user flow with these two things in mind:

1. The instructions for the most common type of user (docker) should remain really simple
2. It should be easy to switch to podman version of the installation instructions - ideally there is a single page to both install earthly and configure podman, rather than two different pages that the user needs to go through.

File: .gitbook.yaml [hunk 3]
```diff
@@ -20,6 +20,7 @@ redirects:
     guides/shared-cache: guides/shared-cache.md
     guides/target-ref: guides/target-ref.md
     guides/udc: guides/udc.md
+    guides/podman: guides/podman.md
```
[the remark this answers is not in the document]
Comment (r-LaForge) [reply .gitbook.yaml]: removed

**Comment by r-LaForge** [comment]
@vladaionescu 
Agreed. I was thinking I'd add this guide as a starting point and then adjust the documentation accordingly.
Would you recommend instead updating all of the spots in a single PR?

I've already combed through and found a number of spots I'd like to update.

**Comment by vladaionescu** [comment]
> Would you recommend instead updating all of the spots in a single PR?

Separate PR makes sense too! No opinion either way!

**Comment by vladaionescu** [comment]
You might want to take a look at this page too: https://github.com/earthly/website/blob/main/website/get-earthly.md

**Review by vladaionescu — APPROVED** [review]

File: docs/guides/podman.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
+# Podman
+[Podman](https://podman.io/) is an alternative to docker; 
+it's a daemonless container engine for developing, managing and running OCI containers on a Linux system.
+Podman also works on Mac using a [podman machine](https://docs.podman.io/en/latest/markdown/podman-machine.1.html).
+
+## Prerequisites
+ - [Install podman](https://podman.io/getting-started/installation)
+ - Mac: ensure a [podman machine](podman machine) is running.
```
Comment (adamgordonbell) [inline docs/guides/podman.md:8]: Not sure this link works, unless its some git book magic I'm not aware of

**Review by adamgordonbell — COMMENTED** [review]
Looks good. One small link issue.

**Pull request merged**